src/duk_debugger.c

```diff
@@ -1639,6 +1684,47 @@ DUK_LOCAL void duk__debug_process_message(duk_hthread *thr) {
 	return;
 }
 
+/* Halt execution and enter a debugger message loop until execution is resumed
+ * by the client. PC for the current activation may be temporarily decremented
+ * so that the "current" instruction will be shown by the client.
+ */
+
+DUK_INTERNAL void duk_debug_halt_execution(duk_hthread *thr, duk_bool_t use_prev_pc) {
+	duk_activation *act;
+	duk_hcompiledfunction *fun;
+	duk_instr_t *old_pc;
+
+	DUK_HEAP_SET_PAUSED(thr->heap);
+
+	act = duk_hthread_get_current_activation(thr);
+	fun = (duk_hcompiledfunction *) DUK_ACT_GET_FUNC(act);
```

Comment:
Related to the SyntaxError issue: I guess there's actually no guarantee that `fun` will be a `duk_hcompiledfunction` here if the error is thrown from C code? That doesn't seem to cause issues below: the `act->curr_pc` check checks for NULL first, and a non-NULL `act->curr_pc` implies `fun` is a compiled function. But reading this one might expect `fun` to be a compiledfunction.